### PR TITLE
SONARJAVA-3228 Handle NPE when resolving super type in interface

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

**Comment by m-g-sonar** [comment]
@amelie-renard-sonarsource @guillaume-dequenne-sonarsource please do NOT squash the PR when merging.

File: java-checks/src/test/java/org/sonar/java/filters/FilterVerifier.java [hunk 1]
```diff
@@ -143,4 +150,23 @@ public void visitTrivia(SyntaxTrivia syntaxTrivia) {
...
+    context.setSettings(new MapSettings().setProperty("sonar.java.failOnException", true));
+    SonarComponents sonarComponents = new SonarComponents(null, context.fileSystem(), null, null, null) {
+      @Override
+      public boolean reportAnalysisError(RecognitionException re, InputFile inputFile) {
+        return false;
+      }
+
+      @Override
+      public void addAnalysisError(AnalysisError analysisError) {
+        Fail.fail(String.format("Should not fail analysis (%s)", analysisError.getKind().name()));
```
Comment (guillaume-dequenne-sonarsource) [inline java-checks/src/test/java/org/sonar/java/filters/FilterVerifier.java]: Maybe there is something I'm missing but I feel like there should be a test case triggering this. Currently I can just comment this change and all tests are passing anyway. Even when leaving out your fix on the other commit, I've been unable to reach this (I'm pretty sure I'm doing something wrong here - but I still think we should have an easier way to make sure this does its intended job).

Comment (m-g-sonar) [reply java-checks/src/test/java/org/sonar/java/filters/FilterVerifier.java]: Added a test case.

**Comment by amelie-renard-sonarsource** [comment]
LGTM

**Pull request merged**